src/test/csfle.rs

```diff
@@ -107,25 +96,17 @@ async fn custom_key_material() -> Result<()> {
     let _guard = LOCK.run_exclusively().await;
 
     let (client, datakeys) = init_client().await?;
-    let enc = ClientEncryption::new(
-        ClientEncryptionOptions::builder()
-            .key_vault_client(client.into_client())
-            .key_vault_namespace(KV_NAMESPACE.clone())
-            .kms_providers(LOCAL_KMS.clone())
-            .build(),
-    )?;
+    let enc = ClientEncryption::builder(client.into_client(), KV_NAMESPACE.clone())
+        .kms_provider(KmsProvider::Local, LOCAL_KMS.clone())
+        .build()?;
 
     let key = base64::decode(
         "xPTAjBRG5JiPm+d3fj6XLi2q5DMXUS/f1f+SMAlhhwkhDRL0kr8r9GDLIGTAGlvC+HVjSIgdL+RKw\
          ZCvpXSyxTICWSXTUYsWYPyu3IoHbuBZdmw2faM3WhcRIgbMReU5").unwrap();
     let id = enc
-        .create_data_key(
-            &KmsProvider::Local,
-            DataKeyOptions::builder()
-                .master_key(MasterKey::Local)
-                .key_material(key)
-                .build(),
-        )
+        .create_data_key(MasterKey::Local)
```

Comment:
Examples like this are, I think, a pretty strong argument in favor of going this way.

Reply:
It definitely is way less verbose, which is a huge win. I also imagine the integration with rust-analyzer is much better too. So while overall it does seem to be a better user experience, it is a pretty big departure from what the rest of our API currently looks like, which concerns me a bit. @kmahar @isabelatkinson what do you all think?